src/main/java/org/truffleruby/language/objects/MetaClassNode.java

```diff
@@ -103,7 +103,8 @@ protected RubyClass metaClassObject(RubyDynamicObject object) {
     @Specialization(guards = "isForeignObject(object)")
     protected RubyClass metaClassForeign(Object object,
             @CachedContext(RubyLanguage.class) RubyContext context) {
-        return context.getCoreLibrary().truffleInteropForeignClass;
+        // return context.getCoreLibrary().truffleInteropForeignClass;
+        return context.getCoreLibrary().polyglotForeignObjectClass;
```

Comment:
This needs to return Polyglot::ForeignObject or Polyglot::ForeignArray depending on `hasArrayElements()`.
Same for LogicalClassNode.

Reply:
OK. I'll change that.

We talked about making this dynamic.  Is it possible? Could this decision be left to ruby and not be hardcoded here?

Reply:
For now, keep it simple and have it here.